lib/blueprinter/optimizer.rb

```diff
@@ -0,0 +1,36 @@
+module Blueprinter
+  class Optimizer
+    def initialize(object)
+      @object = object
+    end
+
+    def optimize!(fields)
+      return object unless active_record_relation?
+      select_columns = (active_record_attributes & fields.map(&:method)) +
+                       required_lookup_attributes
+      object.select(*select_columns)
+    end
+
+    private
+
+    attr_reader :object
+
+    def active_record_relation?
+      !!defined?(ActiveRecord::Relation) &&
```

Comment:
Any reason we're doing `!!defined?` instead of just `defined?`?

Reply:
I think `defined?` is strange in that it does not return `true` or `false`. It either returns a String or nil. Generally methods with `?` should return a boolean.

If we do NOT force it into a boolean with `!!` and we are using Blueprinter on a non activerecord project, this method, `active_record_relation?`, would return `nil` instead of `false`.

Reply:
Do we care about anything more than falsey?